content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java

```diff
@@ -1992,8 +1998,25 @@ private synchronized void doDragDropUpload(HttpServletRequest request, HttpServl
 
 				if (collection!=null)
 				{
-					logger.debug("Adding resource "+uploadFileName+" in collection "+collection.getId());
-					resource = ContentHostingService.addResource(collection.getId(), Validator.escapeResourceName(basename),Validator.escapeResourceName(extension),5);
+					//get the resourceId by using collectionName and uploadFileName
+					resourceId = collectionName +"/"+ uploadFileName;
+					try{
+						//check if resource in collection exists
+						ContentHostingService.getResource(resourceId);
+						//if user has chosen to overwrite existing resource save the new copy
+						if(overwrite){
```

Comment:
Ok but by doing that you've included more code as a result anyway  i.e, line 1886, and the extra parameter, and you've now got 2 parameters that are called pretty much the same - moreover the extra parameter in the method makes it harder to read (and maintain).  Instead you could just have this one line twice: request.getParameter("overwrite")!= null && request.getParameter("overwrite").equals("true")'  or else move lines 1903-6 to line 2006